proxy/src/main/java/com/velocitypowered/proxy/connection/client/LoginSessionHandler.java

```diff
@@ -289,4 +305,8 @@ public void disconnected() {
       connectedPlayer.teardown();
     }
   }
+
+  private void disconnect(String username, LoginFailEvent.Reason reason){
```

Comment:
When I suggested this function name, I also wanted it to take over the responsibility of also disconnecting the user, not just firing the event. Please fix this or change the misleading function name.